### DISCUSSION TITLE
After creating MR cant get actual MR status info

### DISCUSSION THREAD

**Discussion by dimshush** [python-gitlab/python-gitlab#2131] [Q&A] [answered]
## Description of the problem, including code/CLI snippet

Hey guys

Im creating MR with python-gitlab. After creation i need to check for mr_conflicts and pipeline state.

    ```
    project = gl.projects.get(PROJECT_ID, lazy=False)
    create_mr = project.mergerequests.create({'id': PROJECT_ID,
                                          'title': 'Testing multiMR',
                                          'source_branch': SOURCE_BRANCH,
                                          'target_branch': TARGET_BRANCH,
                                          'labels': [LABEL]})
    mr_iid = getattr(create_mr, 'iid')
    time.sleep(3)
    get_mr = project.mergerequests.get(mr_iid)
    mr_conflict = getattr(get_mr, 'has_conflicts')
    mr_pipeline = getattr(get_mr, 'pipeline', 'No pipeline')
    ```

But I dont get actual info with this requests! I have answer with data like MR was created just now and has no info about coflicts or pipelines.

Same time if i start another script from different console only for checking MR status i got right actual information.

Another moment. If in the script with MR creation i run cycle

     ```
     n = 4
     while n > 0: 
      
      time.sleep(1)
      n -= 1
      project.refresh()
      get_mr = project.mergerequests.get(mr_iid)
      mr_conflict = getattr(get_mr, 'has_conflicts')
      mr_pipeline = getattr({get_mr}, 'pipeline', 'No pipeline')
      ```

still no updated info.

How can i get actual info about MR inside script?

## Expected Behavior

get actual info about MR inside creation MR script

## Actual Behavior

I get data like MR was created just now and no updates in its state

## Specifications

  - python-gitlab version:  v3.6.0
  - API version you are using (v4): v4
  - Gitlab server version (or gitlab.com): Community Edition 14.6.2

**Comment by JohnVillalovos** [comment]
My initial thought is that the Gitlab server needs more time. It takes time from when the MR is created until the Gitlab server has figured out if the MR has conflicts. For a test of this I would keep retrying to get the status for at least two minutes. If during two minutes that status does not get updated then that would seem strange. But if it does get updated within that time then you will need to figure out how long is a safe amount of time to wait to make sure you get the updated information.

**Comment by dimshush** [comment, marked as answer]
In fact my cycle worked fine and status updated after 2-3 secs
I was confused by output of this string
mr_pipeline = getattr({get_mr}, 'pipeline', 'No pipeline')

**Reply by JohnVillalovos** [reply]
> In fact my cycle worked fine and status updated after 2-3 secs I was confused by output of this string mr_pipeline = getattr({get_mr}, 'pipeline', 'No pipeline')

Glad to hear you got it working.

I didn't notice before the fact that `get_mr` was inside of `{}`.
